### DISCUSSION TITLE
Easy to use Jupyter Notebook for Youtube video inference 📺

### DISCUSSION THREAD

**Discussion by ArthurFDLR** [openai/whisper#239] [Show and tell]
[![notebook shield](https://img.shields.io/static/v1?label=&message=Notebook&color=blue&style=for-the-badge&logo=googlecolab&link=https://colab.research.google.com/github/ArthurFDLR/whisper-youtube/blob/main/whisper_youtube.ipynb)](https://colab.research.google.com/github/ArthurFDLR/whisper-youtube/blob/main/whisper_youtube.ipynb) [![repository shield](https://img.shields.io/static/v1?label=&message=Repository&color=blue&style=for-the-badge&logo=github&link=https://github.com/ArthurFDLR/whisper-youtube)](https://github.com/ArthurFDLR/whisper-youtube)

I've made a simple Jupyter Notebook including Colab forms to ease Whisper inference on Youtube videos and save the result on your Google Drive.
This is mainly meant for non-technical folks, but the parameter selection GUI is also very useful for more advanced use cases and fine-tuned inference experimentation.

**Comment by dgoryeo** [comment]
Hi Arthur, 

Great job on mking additional parameters available (in the UI). Can this be used for local files too? 

Thanks

**Reply by ArthurFDLR** [reply]
I'm glad you find the Notebook helpful!
I've just modified the video selection to load media from your Google Drive. You should be able to use any audio/video format supported by `FFMPEG`, but I've only tested `mp4` and `wav`.

**Reply by dgoryeo** [reply]
Thanks. Will test it shortly. I usually extract mp3 from a video for transcription. I'm not sure if Whisper may have any preference in loss-less wav, vs mono-mp3.

**Reply by dgoryeo** [reply]
Quick update: mounting Google Drive works well. Also, I was able to run the script on an mp3 file. Unfortunately, although I had selected srt output, it only produced plain text --no timestamps. 

The next task for me is to learn all the parameters. Thanks!

**Reply by ArthurFDLR** [reply]
Thanks for the input. I fixed the issue with the `.srt` outputs; there was a typo in my code.

**Reply by dgoryeo** [reply]
Thanks. I'll give it a try.

**Reply by dgoryeo** [reply]
Hi, I think I am missing a settings in my run.  I just tried the notebook on a long (local file) video, ca 2hrs long. I did not change any of the parameters and I used medium model. The result was quite poor. The transcription output was repetition of a single line throughout the transciption. The single line was detected at 4th second of the video and then repeated for 493 times over 2 hours. There must be a paremeter/flag/setting that I have gotten wrong but I can't figure out which one.

**Reply by Abdurrafey-Siddiqui** [reply]
@ArthurFDLR 
How can I transcribe multiple vids? Can you add that to the colab code please.

**Comment by ghost** [comment]
can you add playlist transcript please?

**Comment by ArthurFDLR** [comment]
Hey @Venomous912 and @toprak, thanks for the input. I just updated the Notebook to support batch processing:
- All videos within a Youtube playlist;
- All video/audio files in a Google Drive folder;

**Reply by ghost** [reply]
thank you so match

**Comment by dgoryeo** [comment]
Have you considered adding a VAD to this --like Silero?

**Comment by ghost** [comment]
Also, naming output as same as videos would be great

**Comment by MaximeRivest** [comment]
you might like [jupyter-whisper](https://github.com/MaximeRivest/jupyter-whisper) :D
